src/containers/Experiment/DataSetStats.js

```diff
@@ -1,52 +1,120 @@
+import intersect from 'lodash/intersection';
+import union from 'lodash/union';
+import difference from 'lodash/difference';
+
 /**
- * Contains helper methods to calculate stats between a dataset and a list of samples
- * eg, how many processed samples are already in the dataset
+ * Contains several helper methods to perform operations on datasets. A dataset is a dictionary where
+ * the keys are experiment accession codes, and the values are a list of sample accession codes.
  */
-export default class DataSetStats {
-  constructor(dataSet, samples) {
-    this.dataSet = dataSet;
-    this.samples = samples;
+export class DataSetOperations {
+  /**
+   * Intersect two datasets
+   * @param {*} d1 dataset 1
+   * @param {*} d2 dataset 2
+   */
+  static intersect(d1, d2) {
+    const result = {};
+    const addedExperimentAccession = intersect(
+      Object.keys(d1),
+      Object.keys(d2)
+    );
+    for (let experimentAccession of addedExperimentAccession) {
+      result[experimentAccession] = intersect(
+        d1[experimentAccession],
+        d2[experimentAccession]
+      );
+    }
+    return result;
   }
 
-  anyProcessedSamples() {
-    return this.samples.some(x => x.is_processed);
+  /**
+   * Returns true if both given arguments have the same values, without taking into cosideration their order
+   * eg:
+   * sameItems([1,2], [2,1]) === true
+   * @param {Array} s1
+   * @param {Array} s2
+   */
+  static _sameItems(s1, s2) {
+    return difference(s1, s2).length === 0 && difference(s2, s1).length === 0;
+  }
+
+  static equal(d1, d2) {
+    const d1Keys = Object.keys(d1);
+    const d2Keys = Object.keys(d2);
+    if (!DataSetOperations._sameItems(d1Keys, d2Keys)) {
+      // d1 and d2 don't have the same accession codes
+      return false;
+    }
+
+    for (let accession of d1Keys) {
```

Comment:
If this function may be used in other scenarios,  you probably should use a more generic variable name (such as `element` or `entry`) than `accession`.

Reply:
Maybe we can change the function name, or add comments to make this more clear. 

This function it's only meant to be used to compare `dataSets`. Which are objects, where the keys are experiment accession codes and the values are lists of sample accession codes (It's the same value we send on `data` for the `/dataset` endpoint). That's why I named this variable `accession`, to try to make it more explicit to what it was referring to.

Do you think this makes sense?

Reply:
Since the function is only used to compare datasets, using `accession` is acceptable. I just want to clarify.